source/beamline/BioXAS/BioXASZebraInput.cpp

```diff
@@ -0,0 +1,74 @@
+#include "BioXASZebraInput.h"
+
+#include "beamline/BioXAS/BioXASZebraCommands.h"
+
+BioXASZebraInput::BioXASZebraInput(const QString &name, const QString &baseName, QObject *parent) :
+	AMControl(name, "", parent)
+{
+	connected_ = false;
+
+	valueControl_ = new AMSinglePVControl(
+				QString("InputControlValue"),
+				QString("%1").arg(baseName),
+				this,
+				1.0);
+
+	statusControl_ = new AMReadOnlyPVControl(
+				QString("InputControlStatus"),
+				QString("%1:STA").arg(baseName),
+				this);
+
+	allControls_ = new AMControlSet(this);
+	allControls_->addControl(valueControl_);
+	allControls_->addControl(statusControl_);
+
+	connect(allControls_, SIGNAL(connected(bool)), this, SLOT(onControlSetConnectedChanged(bool)));
+	connect(valueControl_, SIGNAL(valueChanged(double)), this, SLOT(onInputValueChanged()));
+	connect(statusControl_, SIGNAL(valueChanged(double)), this, SLOT(onInputStatusChanged()));
+}
+
+BioXASZebraInput::~BioXASZebraInput()
+{
+
+}
+
+int BioXASZebraInput::inputValue() const
+{
+	return int(valueControl_->value());
+}
+
+QString BioXASZebraInput::inputValueString() const
+{
+	return BioXASZebraCommand::nameFromCommand(inputValue());
+}
+
+bool BioXASZebraInput::inputStatus() const
```

Comment:
This looks weird to me. When I see "xxx status", I'm expecting something like state instead of boolean value. Recommend to have a function mapping with the "1" status.